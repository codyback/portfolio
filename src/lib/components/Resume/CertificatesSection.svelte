<script lang="ts">
	import { getShortDate } from '@/lib/utils/DateUtils';

	import type { Certificate, TechnologyCertificate } from '@prisma/client';

	type CertificateWithAssociations = TechnologyCertificate & {
		certificates: Certificate[];
	};

	export let certifications: CertificateWithAssociations[];
</script>

<section id="cvCertificates">
	<div class="border-b-2 border-black pb-3">
		<h3 class="uppercase text-2xl font-bold text-right">Certificates</h3>
	</div>

	<div class="border-t-4 border-black mt-0.5 pt-8 font-bold">
		{#each certifications as parent, i}
			<div class="flex items-center">
				<span class="text-lg">{parent.name}</span>
				<span class="px-4 py-0.5 ml-2 bg-black text-white rounded-full text-xs">
					{parent.certificates.length}
				</span>
			</div>
			<hr class="border border-black my-4" />
			{#each parent.certificates as certificate, j}
				<div class="grid grid-cols-2 items-center">
					<p class="sm:col-span-1 text-lg leading-none">
						{certificate.name}
					</p>
					<div class="col-span-1 text-right">
						<span
							class="px-4 py-0.5 border border-black rounded-full text-xs uppercase"
						>
							{getShortDate(certificate.completedAt)}
						</span>
					</div>
				</div>
				{#if j + 1 !== parent.certificates.length}
					<hr />
				{/if}
			{/each}
			{#if i + 1 !== certifications.length}
				<hr class="border border-black my-4" />
			{/if}
		{/each}
	</div>
</section>
