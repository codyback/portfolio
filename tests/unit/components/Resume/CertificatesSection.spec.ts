import CertificatesSection from '@/lib/components/Resume/CertificatesSection.svelte';
import type { Certificate, TechnologyCertificate } from '@prisma/client';
import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';

const certifications: (TechnologyCertificate & {
	certificates: Certificate[];
})[] = [
	{
		id: 'technology-cert-id',
		name: 'testing-title',
		homepageId: 'homepage-id',
		certificates: [
			{
				id: 'cert-id',
				name: 'testing',
				completedAt: new Date(),
				technologyCertificateId: '',
			},
		],
	},
];

describe('CertificatesSection', () => {
	it('renders correctly with open false', () => {
		const { container } = render(CertificatesSection, { certifications });

		expect(container).toMatchSnapshot();
	});
});
