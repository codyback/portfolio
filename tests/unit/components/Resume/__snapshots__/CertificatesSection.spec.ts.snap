// Vitest Snapshot v1

exports[`CertificatesSection > renders correctly with open false 1`] = `
<body>
  <div>
    <section
      id="cvCertificates"
    >
      <div
        class="border-b-2 border-black dark:border-neutral-500 pb-3"
      >
        <h3
          class="uppercase text-2xl font-bold text-right"
        >
          Certificates
        </h3>
      </div>
       
      <div
        class="border-t-4 border-black dark:border-neutral-500 mt-0.5"
      />
      <!--&lt;SectionHeading&gt;-->
       
      <div
        class="pt-8 font-bold"
      >
        <div
          class="flex items-center"
        >
          <span
            class="text-lg"
          >
            testing-title
          </span>
           
          <span
            class="px-6 py-0.5 ml-2 bg-black dark:bg-neutral-300 dark:text-dark-900 text-white rounded-full font-extrabold text-xs"
          >
            1
          </span>
        </div>
         
        <hr
          class="border dark:border-neutral-500 border-black my-4"
        />
         
        <div
          class="grid grid-cols-2 items-center"
        >
          <p
            class="sm:col-span-1 text-lg leading-none"
          >
            testing
          </p>
           
          <div
            class="col-span-1 text-right"
          >
            <span
              class="px-4 py-0.5 border border-black dark:border-neutral-500 rounded-full text-xs uppercase"
            >
              Jan 2023
            </span>
          </div>
        </div>
         
        
         
        
      </div>
    </section>
    <!--&lt;CertificatesSection&gt;-->
  </div>
</body>
`;
